### PR TITLE
fix(api,repositories): clean not pushed branch when checkout

### PR DESCRIPTION
**Pull request by richardlt**
1. Description
Fix repo issue and show request id in panic recovery log.

1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#13577.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#13578.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Docs ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘  
  * Standalone tests 1 -   
  * Standalone tests 2 -   
  * Standalone tests 3 -

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#13578.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Docs ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘  
  * Standalone tests 1 -   
  * Standalone tests 2 -   
  * Standalone tests 3 -

File: engine/api/router.go [hunk 1]
```diff
@@ -106,6 +106,30 @@ func (r *Router) compress(fn http.HandlerFunc) http.HandlerFunc {
...
+func (r *Router) setRequestID(h http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var requestID string
+		if existingRequestID := r.Header.Get(requestIDHeader); existingRequestID != "" {
+			if _, err := uuid.FromString(existingRequestID); err == nil {
+				requestID = existingRequestID
+			}
+		}
+		if requestID == "" {
+			requestID = uuid.NewV4().String()
```
Comment (fsamin) [inline engine/api/router.go:120]:
```suggestion
			requestID = sdk.UUID()
```

File: engine/repositories/processor_checkout.go [hunk 2]
```diff
@@ -14,7 +14,8 @@ func (s *Service) processCheckout(ctx context.Context, op *sdk.Operation) error
 	}
 	log.Debug("processCheckout> repo cloned with current branch: %s", currentBranch)
 
-	if err := gitRepo.ResetHard("origin/" + currentBranch); err != nil {
+	// Clean no commited changes if exists
+	if err := gitRepo.ResetHard("HEAD"); err != nil {
 		return sdk.WithStack(err)
 	}
 	log.Debug("processCheckout> repo reset to origin/%s", currentBranch)
```
Comment (fsamin) [inline engine/repositories/processor_checkout.go:21]:
```suggestion
	log.Debug("processCheckout> repo reset to HEAD")
```

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#13595.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Docs ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘  
  * Standalone tests 1 -   
  * Standalone tests 2 -   
  * Standalone tests 3 -

**Pull request merged**